vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -995,4 +995,36 @@ public Style getStyle() {
                 "Dialog does not support adding styles to overlay");
     }
 
+    /**
+     * Close all the Dialog components associated within the current UI.
+     */
+    public static void closeAllDialogs() {
+        if (UI.getCurrent() == null || !UI.getCurrent().isAttached()) {
+            throw new IllegalStateException(
+                    "Can't remove Dialogs when current UI is null or detached.");
+        }
+        UI.getCurrent().getChildren().forEach(component -> {
```

Comment:
I'm wondering is this even a valid usage of Dialog component? Examples only show Dialog.open() calls which also makes more sense.

Reply:
Unfortunately, Dialog is still a `Component` so it's possible to add it inside a component container. For example, if you have a (non-modal) Dialog, you can place another dialog inside to have it automatically closed when the host dialog closes.

Internal "auto-attach" logic adds a `Dialog` to the `UI` if no parent is otherwise provided.

Reply:
As you see, I do not even promise that in JavaDoc

Reply:
If the API is called `closeAllDialogs()`, it assumably should also close open dialogs that are not direct siblings of the UI.

Reply:
I also considered adding the method to UI, but our build architecture makes it complex. But being method in the UI, this limitation would be intuitively clearer. Technically it could also be possible to recurse the entire component tree ... But that has performance impacts.